services/graphite/README.md

```diff
@@ -0,0 +1,124 @@
+
+The special value _measurement_ is used to define the measurement name.  It can have a trailing `*` to indicate that the remainder of the metric should be used.  If a _measurement_ is not specified, the full metric name is used.
+
+### Basic Matching
+
+`servers.localhost.cpu.loadavg.10`
+* Template: `.host.resource.measurement*`
+* Output:  _measurement_ =`loading.10` _tags_ =`host=localhost resource=cpu`
+
+### Multiple Measurement Matching
+
+The _measurement_ can be specified multiple times in a template to provide more control over the measurement name.  Multiple values
+will be joined together using the _Separator_ config variable.  By default, this value is `.`.
+
+`servers.localhost.cpu.cpu0.user`
+* Template: `.host.measurement.cpu.measurement`
+* Output: _measurement_ = `cpu.user` _tags_ = `host=localhost cpu=cpu0`
+
+Since '.' requires queries on measurements to be double-quoted, you may want to set this to `_` to simplify querying parsed metrics.
+
+`servers.localhost.cpu.cpu0.user`
+* Separator: `_`
+* Template: `.host.measurement.cpu.measurement`
+* Output: _measurement_ = `cpu_user` _tags_ = `host=localhost cpu=cpu0`
+
+### Adding Tags
+
+Additional tags can be added to a metric that don't exist on the received metric.  You can add additional tags by specifying them after the pattern.  Tags have the same format as the line protocol.  Multiple tags are separated by commas.
+
+`servers.localhost.cpu.loadavg.10`
+* Template: `.host.resource.measurement* region=us-west,zone=1a`
+* Output:  _measurement_ = `loading.10` _tags_ = `host=localhost resource=cpu region=us-west zone=1a`
+
+## Multiple Templates
+
+One template may not match all metrics.  For example, using multiple plugins with diamond will produce metrics in different formats.  If you need to use multiple templates, you'll need to define a prefix filter that must match before the template can be applied.
+
+### Filters
+
+Filters have a similar format to templates but work more like wildcard expressions.  When multiple filters would match a metric, the more specific one is chosen.  Filters are configured by adding them before the template.
+
+For example,
+
+```
+servers.localhost.cpu.loadavg.10
+servers.host123.elasticsearch.cache_hits 100
+servers.host456.mysql.tx_count 10
+```
+* `servers.*` would match all values
+* `servers.*.mysql` would match `servers.host456.mysql.tx_count 10`
+* `servers.localhost.*` would match `servers.localhost.cpu.loadavg`
+
+## Default Templates
+
+If no template filters are defined or you want to just have one basic template, you can define a default template.  This template will apply to any metric that has not already matched a filter.
+
+```
+dev.http.requests.200
+prod.myapp.errors.count
+dev.db.queries.count
+```
+
+* `env.app.measurement*` would create
+  * _measurement_=`requests.200` _tags_=`env=dev,app=http`
```

Comment:
When used in the template, the should be comma separated.  The tags should be separated from the template which a space though.